data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/PitWorker.java

```diff
@@ -138,9 +134,14 @@ private void processIndex(final SourcePartition<OpenSearchIndexProgressState> op
                         .withPaginationSize(searchConfiguration.getBatchSize())
                         .withSearchAfter(Objects.nonNull(searchPointInTimeResults) ? searchPointInTimeResults.getNextSearchAfter() : null)
                         .build());
-                buffer.writeAll(searchPointInTimeResults.getDocuments().stream().map(Record::new).collect(Collectors.toList()), BUFFER_TIMEOUT_MILLIS);
-            } catch (final TimeoutException e) {
-                // todo: implement backoff and retry, can reuse buffer accumulator code from the s3 source
+
+                searchPointInTimeResults.getDocuments().stream().map(Record::new).forEach(record -> {
+                    try {
+                        bufferAccumulator.add(record);
+                    } catch (Exception e) {
+                        LOG.error("Failed writing OpenSearch documents to buffer due to: {}", e.getMessage());
```

Comment:
Should we emit a metric here and/or log the document that failed? The user won't be able to take action without knowing which documents are failing

Reply:
Agreed but can we do that in the future? I’ll create an issue for it

Reply:
Will the message not contain the failed record info? Otherwise how do I get it. Log it in the `BufferAccumulator`?

Reply:
Looking back at the code, I think my comment was wrong. BufferAccumulator::add adds to the local collection and flushes if needed. There might be a case where adding to the local collection fails, but this exception should almost always be because the flush failed. I thought we were silently dropping the record if this path occurred, but that's not the case.

Having the doc/index in the log is useful to denote progress nonetheless